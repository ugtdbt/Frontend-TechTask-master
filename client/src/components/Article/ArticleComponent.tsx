import React, { memo } from "react";
import type { FC } from "react";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import IconButton from "@mui/material/IconButton";
import FavoriteIcon from "@mui/icons-material/Favorite";
import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
import ShareIcon from "@mui/icons-material/Share";
import CardHeader from "@mui/material/CardHeader";
import { Price } from "../Price";

interface ArticleProps {
  imageUrl: string;
  name: string;
  variantName: string;
  price: number;
}

const Article: FC<ArticleProps> = memo((props: ArticleProps) => {
  const { imageUrl, name, variantName, price } = props;
  return (
    <Card sx={{ maxWidth: 300, width: 280 }} data-testid="article">
      <CardHeader
        title={<span data-testid="article-name">{name}</span>}
        subheader={<span data-testid="article-variantName">{variantName}</span>}
        sx={{ minHeight: 160, alignItems: "start" }}
      />
      <CardMedia
        data-testid="article-image"
        sx={{ width: 200, height: 200, marginX: "auto" }}
        image={imageUrl}
        title={name}
      />
      <CardContent>
        <Price price={price} />
      </CardContent>
      <CardActions>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
        <IconButton aria-label="share">
          <ShareIcon />
        </IconButton>
        <IconButton aria-label="share">
          <ShoppingCartIcon />
        </IconButton>
      </CardActions>
    </Card>
  );
});

Article.displayName = "Article";
export default Article;
