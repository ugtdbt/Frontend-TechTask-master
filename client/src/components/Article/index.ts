export { default as Article } from "./ArticleComponent";
