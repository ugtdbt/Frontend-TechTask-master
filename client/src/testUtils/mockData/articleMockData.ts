export const article = {
  imageUrl:
    "https://cdn1.home24.net/images/media/catalog/product/200x200/png/l/a/lattenrost-super-fix-90-x-200cm-4448116.webp",
  name: "Lattenrost Super Fix",
  variantName: "Breite: 140 cm",
  price: 14990,
};
