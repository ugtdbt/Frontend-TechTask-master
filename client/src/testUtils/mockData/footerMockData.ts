export const footer = {
  description:
    "Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten.",
};
