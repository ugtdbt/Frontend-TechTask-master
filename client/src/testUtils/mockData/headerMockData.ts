export const sections = [
  { title: "furniture", url: "#" },
  { title: "lamps", url: "#" },
  { title: "accessories", url: "#" },
  { title: "textiles", url: "#" },
  { title: "children", url: "#" },
  { title: "kitchen", url: "#" },
  { title: "bath", url: "#" },
  { title: "Garden", url: "#" },
  { title: "Brands", url: "#" },
  { title: "Shop the look", url: "#" },
  { title: "advisory", url: "#" },
  { title: "stores", url: "#" },
  { title: "Sale", url: "#" },
];

export const header = {
  sections,
  title: "Home24",
};
