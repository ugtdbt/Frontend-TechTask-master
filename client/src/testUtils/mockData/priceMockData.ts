export const price = {
  price: 2500,
};
