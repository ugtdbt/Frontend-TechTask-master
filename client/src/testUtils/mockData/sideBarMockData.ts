export const sideBar = {
  list: [
    { name: "Wohnzimmer", urlPath: "wohnzimmer-moebel/" },
    { name: "Polsterm√∂bel", urlPath: "alle-polstermoebel/" },
  ],
};
