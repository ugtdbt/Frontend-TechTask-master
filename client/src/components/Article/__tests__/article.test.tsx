import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { article } from "../../../testUtils/mockData/articleMockData";
import Article from "../ArticleComponent";

describe("Article component", () => {
  const ArticleComponent = (
    <Article
      imageUrl={article.imageUrl}
      name={article.name}
      variantName={article.variantName}
      price={article.price}
    />
  );

  afterEach(() => {
    cleanup();
  });

  it("Article Match Snapshot", () => {
    const tree = renderer.create(ArticleComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render Article component", () => {
    render(ArticleComponent);
    const articleElem = screen.getByTestId("article");
    expect(articleElem).toBeInTheDocument();
  });

  it("Check Article Name", () => {
    render(ArticleComponent);
    const articleName = screen.getByTestId("article-name")?.innerHTML;
    expect(articleName).toContain(article.name);
  });

  it("Check Article variantName", () => {
    render(ArticleComponent);
    const articleVariantName = screen.getByTestId(
      "article-variantName"
    )?.innerHTML;
    expect(articleVariantName).toContain(article.variantName);
  });

  it("Check Article Image", () => {
    render(ArticleComponent);
    const articleImage = screen
      .getByTestId("article-image")
      .getAttribute("style");
    expect(articleImage).toContain(article.imageUrl);
  });
});
