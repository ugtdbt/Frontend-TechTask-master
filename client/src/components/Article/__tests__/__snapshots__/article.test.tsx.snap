// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component Article Match Snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-148xl03-MuiPaper-root-MuiCard-root"
  data-testid="article"
>
  <div
    className="MuiCardHeader-root css-n8ie1k-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
      >
        <span
          data-testid="article-name"
        >
          Lattenrost Super Fix
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
      >
        <span
          data-testid="article-variantName"
        >
          Breite: 140 cm
        </span>
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root css-145s88o-MuiCardMedia-root"
    data-testid="article-image"
    role="img"
    style={
      Object {
        "backgroundImage": "url(\\"https://cdn1.home24.net/images/media/catalog/product/200x200/png/l/a/lattenrost-super-fix-90-x-200cm-4448116.webp\\")",
      }
    }
    title="Lattenrost Super Fix"
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      data-testid="price"
    >
      149,90 €
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      aria-label="add to favorites"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FavoriteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
        />
      </svg>
    </button>
    <button
      aria-label="share"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ShareIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
        />
      </svg>
    </button>
    <button
      aria-label="share"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ShoppingCartIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
